indexd/index/drivers/query/urls.py

```diff
@@ -0,0 +1,75 @@
+from sqlalchemy import func, and_
+
+from indexd.index.drivers.alchemy import IndexRecord, IndexRecordUrl, IndexRecordUrlMetadata
+from indexd.index.drivers.query import URLsQueryDriver
+
+
+driver_query_map = {
+    "sqlite": dict(array_agg=func.group_concat, string_agg=func.group_concat),
+    "postgresql": dict(array_agg=func.array_agg, string_agg=func.string_agg)
+}
+
+
+class AlchemyURLsQueryDriver(URLsQueryDriver):
+    """SQLAlchemy based impl"""
+
+    def __init__(self, alchemy_driver):
+        """ Queries index records based on URL
+        Args:
+            alchemy_driver (indexd.index.drivers.alchemy.SQLAlchemyIndexDriver):
+        """
+        self.driver = alchemy_driver
+
+    def query_urls(self, exclude=None, include=None, only_versioned=True, offset=0, limit=1000):
+
+        with self.driver.session as sxn:
+            # special database specific functions dependent of the selected dialect
+            q_func = driver_query_map.get(sxn.bind.dialect.name)
+
+            query = sxn.query(IndexRecordUrl.did, q_func['string_agg'](IndexRecordUrl.url, ",").label("urls"))\
+                .outerjoin(IndexRecord)
```

Comment:
I don't think outer join is needed here?

Reply:
Without it, you won;t be able to filter by versioned entries. Maybe I can add it only when (un)versioned entries are requested

Reply:
oh right we always need the version column, I was thinking IndexRecord left outer join to IndexRecordUrl might have better performance, but on second thought I think it's the same..